src/host/screenInfo.cpp

```diff
@@ -1981,6 +1990,12 @@ void SCREEN_INFORMATION::UseMainScreenBuffer()
             psiMain->ProcessResizeWindow(&(psiMain->_rcAltSavedClientNew), &(psiMain->_rcAltSavedClientOld));
             psiMain->_fAltWindowChanged = false;
         }
+        else if (_psiMainBuffer->_deferredPtyResize.has_value())
+        {
+            const COORD newSize = _psiMainBuffer->_deferredPtyResize.value().to_win32_coord();
+            _psiMainBuffer->SetViewportSize(&newSize);
+            _psiMainBuffer->_deferredPtyResize = std::nullopt;
+        }

```

Comment:
It's worth noting that the `_fAltWindowChanged` resize is triggered in the `UseAlternateScreenBuffer` as well. I'm not entirely sure why that is, but I would expect this deferred resize to be triggered under the same conditions, unless there's a good reason not to. Of course if we could unify the two somehow then this would be a moot point.

Reply:
should be in 764237c15, thanks for the catch